sui-execution/latest/sui-adapter/src/gas_charger.rs

```diff
@@ -207,6 +207,13 @@ pub mod checked {
             self.gas_status.charge_publish_package(size)
         }
 
+        pub fn charge_upgrade_package(&mut self, size: usize) -> Result<(), ExecutionError> {
+            if self.gas_model_version < 7 {
```

Comment:
Don't you want to use the predicate you defined here?

Reply:
indeed, I just pushed it after realizing the mistake. Great noticing it though!!!!!